File: packages/contract-py/src/tigr_asgi_contract/directions.py
from ._enum_compat import StrEnum


class Direction(StrEnum):
    CLIENT_TO_SERVER = 'client_to_server'
    SERVER_TO_CLIENT = 'server_to_client'
    APP_TO_SERVER = 'app_to_server'
    SERVER_TO_APP = 'server_to_app'
    SYSTEM = 'system'

File: packages/contract-rs/src/directions.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
pub enum Direction {
    #[serde(rename = "client_to_server")]
    ClientToServer,
    #[serde(rename = "server_to_client")]
    ServerToClient,
    #[serde(rename = "app_to_server")]
    AppToServer,
    #[serde(rename = "server_to_app")]
    ServerToApp,
    #[serde(rename = "system")]
    System,
}
